lib/executor/ramping_arrival_rate.go

```diff
@@ -316,25 +316,36 @@ func (varr RampingArrivalRate) Run(parentCtx context.Context, out chan<- stats.S
 	returnedVUs := make(chan struct{})
 	startTime, maxDurationCtx, regDurationCtx, cancel := getDurationContexts(parentCtx, duration, gracefulStop)
 
+	iterations := make(chan struct{})
+	iterators := runningVUs{
+		Iterations: iterations,
+	}
+
 	defer func() {
+		close(iterations)
```

Comment:
I would prefer that you encapsualte the `iterations` as well and have `newRunningVUs` (or IMO `newVUPool`) which make the `iterations` channel and a method that is `stop` to close `iterations` and effectively stop the "running" of vus.